### PR TITLE
Clean up error handling in Secret Manager

### PR DESCRIPTION
**Pull request by qfel**
Clean up the implementation of #334 

Converts errors signaling background worker death to panics. This is generally preferable since SecretManager will not recover from them: all new certificate fetches will fail and no refreshes will be performed ever again.

They are also programming errors: it should not be possible to ever get into that state, save for bugs in SecretManager code (either due to early return in the worker or a panic). While we could try to restart the worker in such circumstances, that seems dangerous as a dead worker signals something is very wrong - better to propagate and panic.

### CODE REVIEW

**Review by hzxuzhonghu — CHANGES_REQUESTED** [review]
It should return error instead of panic

**Comment by qfel** [comment]
> It should return error instead of panic

vs

> Given we cannot recover from errors, this feels a bit like mutex poisoning where it is common practice to panic if it occurs. I am ok with this (despite normally not liking panics) for this case

So, which one?

File: src/identity/manager.rs [hunk 1]
```diff
@@ -355,36 +355,28 @@ impl<T: CertificateProvider> SecretManager<T> {
                 drop(certs);
                 // Notify the background worker to start refreshing the certificate.
                 match self.requests.send((id.clone(), pri)).await {
-                    // Note that this should not happen, the background worker is expected to be
-                    // running.
-                    //
-                    // TODO: Better error propagation.
-                    Err(_) => Err(Error::EmptyResponse(id.clone())),
+                    Err(e) => unreachable!("SecretManager worker died: {e}"),
```
Comment (hzxuzhonghu) [inline src/identity/manager.rs:358]: can you explain why self.requests.send cannot be recovered?

Comment (qfel) [reply src/identity/manager.rs:358]: That means the worker has died, so this and *all* further calls will fail with this error. Basically the SecretManager instance is useless. You can try to recover from it by re-creating the SecretManager, but it's just silencing a programming error in a `try..catch(anything)` fashion. Rust has good support for resource ownership and makes it hard to maintain global state so it's less of a risk, but I have seen this kind of approach cause deadlocks in other languages.

Comment (hzxuzhonghu) [reply src/identity/manager.rs:358]: IC, this is requests send channel.

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**